Sources/SwiftParser/Modifiers.swift

```diff
@@ -59,30 +59,31 @@ extension Parser {
         }
       case (.unowned, _)?:
         elements.append(self.parseUnownedModifier())
-      case (.final, _)?,
-        (.required, _)?,
-        (.optional, _)?,
-        (.lazy, _)?,
-        (.dynamic, _)?,
-        (.infix, _)?,
-        (.prefix, _)?,
-        (.postfix, _)?,
-        (.__consuming, _)?,
-        (.mutating, _)?,
-        (.nonmutating, _)?,
-        (.convenience, _)?,
-        (.override, _)?,
-        (.weak, _)?,
-        (.indirect, _)?,
-        (.isolated, _)?,
-        (.async, _)?,
-        (.nonisolated, _)?,
-        (.distributed, _)?,
-        (._const, _)?,
-        (._local, _)?,
-        (.__setter_access, _)?,
-        (.reasync, _)?:
-        elements.append(self.parseSimpleModifier())
+      case (.final, let handle)?,
```

Comment:
There's a few places where this is the case, but IMO it'd be better to assign kind/handle first and then switch over kind rather than having to do (.kind, let handle) or (.kind, _) everywhere. Not really relevant to this, I just noticed it while going through the changes.

Reply:
The nice thing about this approach is that it guarantees we have a handle if we match a kind without having to force-unwrap it (remember there’s a default case at the bottom that doesn’t have a handle).

Reply:
There's no reason that has to be in the switch though:
```
guard let (kind, handle) = self.at(anyIn: DeclarationModifier.self) else {
  break MODIFIER_LOOP
}

switch kind {
...
}
```

Or am I missing something obvious?

Reply:
No, that would also work. I think that’s just a matter of style now. If there were fewer cases in the `RawTokenKindSubset`, I would prefer a single `switch` instead of the `guard` + `switch` combination. And then I think it’s nice to always use the same style when you’re switching over `RawTokenKindSubset`s.

Reply:
Sure, it's definitely style. I'll win you over eventually!